File: app/src/main/java/com/proxy/shadowsocksr/SSRBackupAgent.java
package com.proxy.shadowsocksr;

import android.app.backup.BackupAgentHelper;
import android.app.backup.SharedPreferencesBackupHelper;

public class SSRBackupAgent extends BackupAgentHelper
{
    private final String HAWK_PREF = "HAWK";

    private final String PREF_BACK_KEY = "com.proxy.shadowsocksr";

    @Override public void onCreate()
    {
        super.onCreate();
        SharedPreferencesBackupHelper spbk = new SharedPreferencesBackupHelper(this, HAWK_PREF);
        addHelper(PREF_BACK_KEY, spbk);
    }
}

File: app/src/main/java/com/proxy/shadowsocksr/SSRBackupAgent.kt
package com.proxy.shadowsocksr

import android.app.backup.BackupAgentHelper
import android.app.backup.SharedPreferencesBackupHelper

class SSRBackupAgent : BackupAgentHelper()
{
    private val HAWK_PREF = "HAWK"

    private val PREF_BACK_KEY = "com.proxy.shadowsocksr"

    override fun onCreate()
    {
        super.onCreate()
        val spbk = SharedPreferencesBackupHelper(this, HAWK_PREF)
        addHelper(PREF_BACK_KEY, spbk)
    }
}
